### PR TITLE
Automated AUR publishing

### PR DESCRIPTION
**Pull request by relu**
Related to #530

### CODE REVIEW

**Comment by relu** [comment]
Too bad we can't execute `makepkg --printsrcinfo` in order to automatically generate the `.SRCINFO` file used to publish the packages.

File: ci/aur/flux-bin/publish.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export SHA256SUM_ARM=$(sha256sum ${ROOT}/dist/flux_${PKGVER}_linux_arm.tar.gz | awk '{ print $1 }')
+export SHA256SUM_ARM64=$(sha256sum ${ROOT}/dist/flux_${PKGVER}_linux_arm64.tar.gz | awk '{ print $1 }')
+export SHA256SUM_AMD64=$(sha256sum ${ROOT}/dist/flux_0.4.2_linux_amd64.tar.gz | awk '{ print $1 }')
+
+envsubst '$PKGVER $PKGREL $SHA256SUM_AMD64 $SHA256SUM_ARM $SHA256SUM_ARM64' < .SRCINFO.template > .pkg/.SRCINFO
+envsubst '$PKGVER $PKGREL $SHA256SUM_AMD64 $SHA256SUM_ARM $SHA256SUM_ARM64' < PKGBUILD.template > .pkg/PKGBUILD
+
+cd .pkg
+git config user.name "GoReleaser"
+git config user.email "goreleaser@goreleaser.com"
```
Comment (relu) [inline ci/aur/flux-bin/publish.sh:39]: Do we want to change these?

File: .goreleaser.yml [hunk 2]
```diff
@@ -50,3 +50,21 @@ brews:
         type: optional
     test: |
       system "#{bin}/flux --version"
+publishers:
```
Comment (relu) [inline .goreleaser.yml:53]: I'm not sure this is configured correctly.

File: .goreleaser.yml [hunk 3]
```diff
@@ -50,3 +50,21 @@ brews:
...
+  - name: aur-pkg-bin
+    ids:
+      - nix
+    cmd: |
+      eval $(ssh-agent -s)
+      ssh-add <(echo "{{ .Env.AUR_SSH_PRIVATE_KEY }}")
```
Comment (relu) [inline .goreleaser.yml:59]: Needs environment variable set.

Comment (hiddeco) [reply .goreleaser.yml:59]: Secret is available in this repository as `AUR_BOT_SSH_PRIVATE_KEY`, configuration for GitHub actions can be found at: https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

File: ci/aur/flux-git/.SRCINFO.template [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	depends = glibc
+	optdepends = kubectl
+	provides = flux-cli
+	conflicts = flux-bin
+	conflicts = flux-go
+	source = git+https://github.com/fluxcd/flux2.git
+	md5sums = SKIP
+
+pkgname = flux-git
+
```
[the remark this answers is not in the document]
Comment (relu) [reply ci/aur/flux-git/.SRCINFO.template:21]: It's how makepkg generates 'em...

File: .goreleaser.yml [hunk 5]
```diff
@@ -50,3 +50,21 @@ brews:
...
     test: |
       system "#{bin}/flux --version"
+publishers:
+  - name: aur-pkg-bin
+    ids:
+      - nix
+    cmd: |
+      eval $(ssh-agent -s)
+      ssh-add <(echo "{{ .Env.AUR_BOT_SSH_PRIVATE_KEY }}")
+      ci/aur/flux-bin/publish.sh {{ .Version }}
```
Comment (hiddeco) [inline .goreleaser.yml:60]: Given all our CI is in/on GitHub, I would maybe just place this in the `.github` directory (`.github/release/aur`?).

File: .goreleaser.yml [hunk 6]
```diff
@@ -50,3 +50,21 @@ brews:
...
+  - name: aur-pkg-git
+    cmd: |
+      eval $(ssh-agent -s)
+      ssh-add <(echo "{{ .Env.AUR_BOT_SSH_PRIVATE_KEY }}")
+      ci/aur/flux-git/publish.sh {{ .Version }}
+  - name: aur-pkg-git
```
Comment (hiddeco) [inline .goreleaser.yml:66]:
```suggestion
  - name: aur-pkg-go
```

File: .goreleaser.yml (hunk 5 above)
Comment (relu) [reply .goreleaser.yml:60]: Makes sense, I was also thinking about where to better place this but `.github` was not what I thought would be the right place. I suppose it should be fine judging by your argument regarding all CI stuff living there.

File: .github/aur/flux-bin/.SRCINFO.template [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+	url = https://fluxcd.io/
+	arch = x86_64
+	arch = armv6h
+	arch = armv7h
+	arch = aarch64
+	license = APACHE
+	optdepends = kubectl
+	provides = flux-cli
+	conflicts = flux-git
+	conflicts = flux-go
```
Comment (hiddeco) [inline .github/aur/flux-bin/.SRCINFO.template:14]: I think these need an update too now that the `PKGBUILD.template` files have changed.

Comment (relu) [reply .github/aur/flux-bin/.SRCINFO.template:14]: :eagle: :eyes:

Ugh, you see my point about wanting to have `makepkg` generate these? :smile:

**Comment by hiddeco** [comment]
Waiting for confirmation of successful manual run before stamping, as that only gives us the GoReleaser changes to worry about next release.

**Comment by hiddeco** [comment]
Packages are all up on https://aur.archlinux.org/packages/?O=0&SeB=m&K=fluxcdbot.

**Review by hiddeco — APPROVED** [review]
LGTM :medal_sports:

File: docs/get-started/index.md [hunk 8]
```diff
@@ -34,6 +34,15 @@ curl -s https://toolkit.fluxcd.io/install.sh | sudo bash
 
 The install script downloads the flux binary to `/usr/local/bin`.
 
+If using Arch Linux, there are three packages published in AUR you can use to
+install `flux`:
+- [flux-bin](https://aur.archlinux.org/packages/flux-bin): install the latest
```
Comment (stefanprodan) [inline docs/get-started/index.md:39]: Let's keep only this option for get started.

**Review by stefanprodan — CHANGES_REQUESTED** [review]

**Pull request merged**